src/views/viewer/ViewerLayout.tsx

```diff
@@ -64,9 +67,9 @@ export const ViewerLayout: React.FC = () => {
 
     // delay scroll to allow transition to finish first
     setTimeout(() => {
-      window.scrollTo(0, 0)
+      window.scrollTo(0, navigationType !== 'POP' ? 0 : scrollPosition.current)
```

Comment:
This doesn't seem to work properly, my scroll position isn't correctly restored:

https://user-images.githubusercontent.com/12646744/144088538-3816019e-582f-435c-9c38-8e13d7f77bde.mov

Couldn't we use native browser scroll restoration instead?